charts/gateway-operator/templates/deployment.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
```

Comment:
This and all other resources should support user-provided labels, to allow, for example, adding custom owership metadata.

The [`kong` labels template](https://github.com/Kong/charts/blob/kong-2.38.0/charts/kong/templates/_helpers.tpl#L28-L37) also includes some stock labels related to chart/release info. Unsure if those are particularly useful in practice, but they probably won't hurt.